clients/js/lib/refresh_strategy.js

```diff
@@ -0,0 +1,46 @@
+const fetch = require('node-fetch');
+
+function checkStatus(res) {
+  if (res.ok) {
+    return res;
+  }
+  throw new Error('Unable to download descriptor file');
+}
+
+const joinPath = (url, path) => {
+  if (url.endsWith('/')) {
+    return url + path;
+  }
+  return [url, path].join('/');
+};
+
+const longPollingRefresh = () => async (url, options) =>
+  fetch(url, options)
+    .then(checkStatus)
+    .then((res) => res.buffer());
+
+const versionBasedRefresh = () => {
+  let prevLatestVersion = 0;
+  return async (url, options) => {
+    const versionsURL = joinPath(url, 'versions');
+    const versions = await fetch(versionsURL, options)
+      .then(checkStatus)
+      .then((res) => res.json())
+      .then((data) => data.versions);
+    const maxVersion = Math.max(...versions);
```

Comment:
Okay, I will add that. Original thought process was, Since it uses stencil versions API, if API returns successful response it will have `versions` field.